internal/handlers/pg/pgdb/database_metadata.go

```diff
@@ -134,7 +134,34 @@ func (ms *metadataStorage) store(ctx context.Context) (tableName string, created
 		return
 	}
 
-	tableName = collectionNameToTableName(ms.collection)
+	// Index to ensure that table name is unique
+	key = IndexKey{{Field: `table`, Order: types.Ascending}}
+	if err = createPgIndexIfNotExists(ctx, ms.tx, ms.db, dbMetadataTableName, dbMetadataIndexName, key, true); err != nil {
+		err = lazyerrors.Error(err)
+		return
+	}
+
+	defaultTableName := collectionNameToTableName(ms.collection)
+
+	var tableNameUnique bool
+	tableName = defaultTableName
+
+	for i := 1; ; i++ {
+		tableNameUnique, err = ms.isTableNameUnique(ctx, tableName)
+		if err != nil {
+			tableName = ""
+			err = lazyerrors.Error(err)
+
+			return
+		}
+
+		if tableNameUnique {
+			break
+		}
+
+		tableName = fmt.Sprintf("%s_%d", defaultTableName, i)
```

Comment:
Do we know is it possible to exceed the PostgreSQL table name limit by renaming and creating collection with a long name and renaming + recreating it multiple times? It doesn't sound like a serious issue although it's nice if we have it in mind :)

Reply:
It seems to me that we have different validation on rename than on create, exceeding the number is even not possible for a <235 x 'b' which wasn't inserted to the collection previously.

```
test> db.runCommand({renameCollection:"test.values2", to:"test.bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" })
MongoServerError: error with target namespace: Invalid collection name: bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
```

As this is very specific corner case we could fix that in followup either :)

Reply:
Yeah, it's possible. Ideally, in this case, we should remove some symbols from the table name... I think I'll create a separate PR about it, as otherwise, this one will be even bigger (especially, with tests).

Reply:
https://github.com/FerretDB/FerretDB/issues/2571

Reply:
Please do that in this issue (if not in this PR). That seems to be a serious problem